### ISSUE TITLE
Separate efficiency standard settings by fuel type in buildings and industry 

### ISSUE THREAD

**Issue by oashmoore** [Energy-Innovation/eps-us#102]
We currently can only model building and industry efficiency by sector and end use. For the state models, we are modelling electricity reduction targets from legislation and IRPs. To achieve the efficiency gains in the electricity sector, we have to reduce other fuels as well. This ends up over-stating the emissions reductions potential of electricity efficiency policies. It also makes it difficult to set EE targets because the legislated electricity savings can exceed the max EE policy lever. 

Separating the efficiency policy setting by fuel type for buildings and industry would help us create policy scenarios that target electricity efficiency.

**Comment by jrissman** [comment]
That's no problem for industry, where the efficiency policy is currently subscripted only by industry category.

In buildings, the efficiency policy is currently subscripted by building type (urban residential, rural residential, commercial) and by end use (heating, cooling, lighting, etc.).  We currently only support adding two subscripts to any single policy lever.  Can we get rid of the subscripting by building type?  That way, it would be set separately by end use and by fuel (electricity, NG, etc.), but would apply to all building types that have that end use using that fuel.

**Comment by mkmahajan** [comment]
Sounds great if we can add the additional fuel type subscript for industry.

In buildings, I'm worried about losing the residential/commercial split since we have received requests to look at specific efficiency programs that only apply to residential buildings. Would we be able to create two different policy levers for residential and commercial buildings (I don't think we would need to differentiate between rural and urban residential), then subscript each by end use and fuel type? I realize that's a little different than how we would ideally set this up though.

**Comment by jrissman** [comment]
That would need three levers, not two.  The urban/rural distinction is important in places like India, even if you do not think it is important in the U.S.

I could potentially work with Todd to add support for a third subscript on policy levers.  But it would make setting these levers miserable in the web app, requiring very many clicks and slider movements.  (The same would be true if we used multiple policy levers.)  We restricted policy levers to two subscripts per lever for a reason - to avoid making the experience of using the EPS terrible for web users.  I fear a triple-subscripted policy would be almost unusable to web users because of the large number of levers to set.

We'll need to rethink the web UI to better handle this.  We have an item already for a "set all" button to help deal with this, but it's probably not enough.  We probably need something like "set all with subscript value X" so you could set all electricity values at once, or all urban residential values at once, or all heating values at once.  We might need to move the subscript setting controls for each policy into its policy setting pane, so only policies - not subscripted elements - would appear in the policy tree.  That would give us more space to use for this new UI and would avoid overwhelming the policy tree (and I don't think we could necessarily overlay buttons or links in the policy tree anyway).

This sort of UI would also let us move the GRA levers for a policy into that policy's own settings pane, potentially.

Can we hold off on implementing the buildings one until we have a reasonable UI design to accommodate it?  (That might be well into 2021.)  I could add the Industry subscript now, if we can put the buildings one on hold until we have a web UI that can appropriately manage our growing need for subscript complexity.

**Comment by jrissman** [comment]
If you urgently need the buildings capability now, we could potentially implement triple-subscript support in the policy tree as a stopgap measure, and accept the user experience will be bad for a while until we get a better UI solution into place.  It's not ideal because it involves work to extend the policy tree functionality in a way we'd later abandon when we roll out the improved subscript setting management UI, but if you need this capability urgently, it may be the least-bad option.

**Comment by mkmahajan** [comment]
I'll defer to Robbie on whether we need to get this in right away vs. waiting.

**Comment by jrissman** [comment]
By the way, this is what the building component efficiency standards lever looks like today, with only two subscripts:

![74771045-53795080-5242-11ea-84e2-5f8fe1184a9e](https://user-images.githubusercontent.com/7120106/97368577-798ed000-1868-11eb-9ba5-f6348a5c7ca3.png)

It's already annoying to set them all in the web app, one at a time.  And there are 10 building fuel types.  Perhaps not all of them need to be exposed in the web app, if a fuel isn't likely to have standards applied or isn't used in a given region, but probably quite a few would be.  So you have to imagine multiplying this number of setting rows by the number of building fuel types commonly in use in a region.  It would get really crazy.  Few people would be willing to go to the trouble of using this policy lever at all, with so many levers to set.

This is why I really prefer to wait for a new UI solution, if we can, before implementing this.

**Comment by jrissman** [comment]
I'm working on implementing the Industry efficiency standards breakout by fuel now.  We'll include this in the next, small model release (which sounds like it will be mostly a data update), but since it won't be data-only, it will release as 3.0.1, not 3.0.0.1

**Comment by jrissman** [comment]
Sounds good.

Subscripting the industry efficiency standards policy by fuel has been completed in 305e113.  I also incremented the version number to 3.0.1 in that commit.

I'm going to close this, because we decided to hold off on the buildings standards policy, as it will likely be obviated by the buildings sector update discussed in issue #92.  I'll make a note in that issue to ensure the standards policy is subscripted by fuel type as part of that item.
